### DISCUSSION TITLE
Hi wojtask! I plan to self-learn crls 4th, want to solve solutions for Chapters 18-35 and share with you.

### DISCUSSION THREAD

**Discussion by sci-42ver** [wojtask/clrs4e-solutions#1590] [General]
I am one undergraduate student graduated last year. I plan to self-learn crls 4th for the future better job. 

Your repo README says you will Migrate Chapters 1-17 and "Add solutions and implementations to Chapters 18-35". Do you mean you will do all Exercises and Problems in Chapters 18-35 from scratch without migration? If so, I think it is fine for me to solve them and share with you although I haven't learnt algorithm before and maybe the solutions have some errors ( Normally, I will ask problems in Stack Exchange when stuck with solving with these problems). If not, then my thoughts may be a bit different from yours about solutions and it is better for you to _do the migration_. I will write my notes in one separate repo of mine for part of the problems in crls (it may be a heavy work to solve all).

I found crls 4th Chapters 18-35 are not totally absent from crls 2nd. So you probably mean migrate all chapters. Do I  understand right?

Do you have any useful advice for me since you have read crls 2nd before? Thanks in advance.

**Comment by wojtask** [comment]
Hi sci-42ver,

Thank you very much for your interest and willingness to help me with this project! As I stated in the README, I already solved chapters 1-17 (and appendices A-C) for CLRS2e, but in Polish (see project [CormenSol](https://github.com/wojtask/CormenSol), now frozen). I consider those solutions as quite reliable, so I don't expect any major bugs or limitations, and I began working on the current project from migrating the solutions from CormenSol.

By _migration_ I understand the two-part process
- adapting the CLRS2e solutions to CLRS4e (keeping them consistent with the problems from the new edition both content- and style-wise),
- translating them to English.

Truth is, I also solved chapter 18 but the solutions are not as solid as those in previous chapters, so additional effort would be required to challenge them and fix any outstanding bugs. The chapters 19-35 I didn't solve in CormenSol so those need to be worked on from scratch.

Please let me know which chapters you are interested in, so I could familiarize myself with the material in the book and so be able to properly review your work. The other areas where you could help are proofreading the content so far (if you're an English native speaker) or implementing algorithms in the [companion project](https://github.com/wojtask/clrs4e-implementations) (if you're familiar with Python).

**Reply by sci-42ver** [reply]
Thanks for your efforts of this project when you are not working.

I will proofread "the content so far".

As you says chapters 18-35 may be needed to solve without direct migration. I will do that with my best efforts although I am not one master. I didn't take participate in contests like ACM as you did. So my solutions may have some errors. Hope our cooperation will be fine.

I began reading this book today. Maybe some time later I will begin read chapter 18-35.

**Reply by wojtask** [reply]
Thank you so much! Please create pull requests once you've found any bugs in the existing solutions. If you can raise any PRs soon or leave any other feedback, I'll make sure to review it and incorporate fixes before release 0.2.

Good luck with reading the book! And don't bother making errors, whenever I made one I learnt from it 😀
